app/src/main/java/com/cyberbot/checkers/ui/activity/GameActivity.kt

```diff
@@ -48,11 +48,11 @@ class GameActivity : AppCompatActivity() {
                 grid.attemptMove(srcEntry, dstEntry)
                 if (dstEntry.player == PlayerNum.SECOND) {
                     val src: GridEntry = grid.filter {
-                        it.player == PlayerNum.FIRST
+                        it.player == PlayerNum.FIRST && grid.calculateAllowedMoves(it, false).isNotEmpty()
                     }.random()
 
                     val dst: GridEntry = grid.filter {
-                        it != src && gridData.moveAllowed(src, it)
+                        it != src && gridData.calculateAllowedMoves(src, false).contains(it)
```

Comment:
Redundant change. The `moveAllowed` method already calls the same statement. Also,

Reply:
Please change this statement back.